src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const obj: GroupsMap<T> = {};
+
+  for (let i = 0; i < items.length; i++) {
+    const value = items[i][key] as string;
```

Comment:
The assumption that `items[i][key]` is always a string may not be correct. If `key` refers to a property that is not a string, this will cause a runtime error. Consider using `String(items[i][key])` to safely convert the value to a string.